### DISCUSSION TITLE
ArduPilot actions all getting stuck

### DISCUSSION THREAD

**Discussion by tridge** [community/community#25675] [Actions]
<p>We’ve been using github actions for the ArduPilot open source autopilot since September 2020. Generally has been very good, but over the last couple of days all our PRs are stuck for a very long time. Here is a typical one stuck for 11 hours:</p><aside class="onebox githubpullrequest" data-onebox-src="https://github.com/ArduPilot/ardupilot/pull/18142">
  <header class="source">

      <a href="https://github.com/ArduPilot/ardupilot/pull/18142" target="_blank" rel="noopener nofollow ugc">github.com/ArduPilot/ardupilot</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ArduPilot/ardupilot/pull/18142" target="_blank" rel="noopener nofollow ugc">AP_Scripting: avoid a error in lua with gcc 10.2 on STM32 with -Werror</a>
    </h4>

    <div class="branches">
      <code>ArduPilot:master</code> ← <code>tridge:pr-lua-warning</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-07-28" data-time="00:33:16" data-timezone="UTC">12:33AM - 28 Jul 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/tridge" target="_blank" rel="noopener nofollow ugc">
          <img alt="tridge" src="https://user-images.githubusercontent.com/831867/181092456-e18c4d0a-52f0-496d-8316-e7a331114786.jpeg" class="onebox-avatar-inline" width="20" height="20">
          tridge
        </a>
      </div>

      <div class="lines" title="1 commits changed 1 files with 3 additions and 0 deletions">
        <a href="https://github.com/ArduPilot/ardupilot/pull/18142/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+3</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">g++ 10.2.1 for STM32 gives this:
![image](https://user-images.githubusercontent<span class="show-more-container"><a href="https://github.com/ArduPilot/ardupilot/pull/18142" target="_blank" rel="noopener nofollow ugc" class="show-more">…</a></span><span class="excerpt hidden">.com/831867/127244840-4e710674-4eca-489b-aacb-48d737dee857.png)
it seems to be a false positive</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/f/0f161fba11fc0f042a7c6bf050f21e6087c8447c.png" data-download-href="/uploads/short-url/29syRJcCKoN8t0MllCYgRpGJYlu.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/831867/181092459-fe81cb43-cec5-4ce6-ad48-ff0c7e382cdd.png" alt="image" data-base62-sha1="29syRJcCKoN8t0MllCYgRpGJYlu" width="690" height="350" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/f/0f161fba11fc0f042a7c6bf050f21e6087c8447c_2_690x350.png, https://aws1.discourse-cdn.com/github/original/3X/0/f/0f161fba11fc0f042a7c6bf050f21e6087c8447c.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/0/f/0f161fba11fc0f042a7c6bf050f21e6087c8447c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/f/0f161fba11fc0f042a7c6bf050f21e6087c8447c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">928×471 72.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
How do we find out why we’re getting stuck?<br>
Other PRs get through almost half the workflows, like this one which has done 28 and has another 28 queued.<aside class="onebox githubpullrequest" data-onebox-src="https://github.com/ArduPilot/ardupilot/pull/18137">
  <header class="source">

      <a href="https://github.com/ArduPilot/ardupilot/pull/18137" target="_blank" rel="noopener nofollow ugc">github.com/ArduPilot/ardupilot</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ArduPilot/ardupilot/pull/18137" target="_blank" rel="noopener nofollow ugc">Logging: added LOG_FILE_RATEMAX parameter</a>
    </h4>

    <div class="branches">
      <code>ArduPilot:master</code> ← <code>tridge:pr-log-rate-max</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-07-27" data-time="08:51:30" data-timezone="UTC">08:51AM - 27 Jul 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/tridge" target="_blank" rel="noopener nofollow ugc">
          <img alt="tridge" src="https://user-images.githubusercontent.com/831867/181092460-acdf18ec-e5ab-4cd9-99d3-fe9487927d61.jpeg" class="onebox-avatar-inline" width="20" height="20">
          tridge
        </a>
      </div>

      <div class="lines" title="20 commits changed 24 files with 302 additions and 117 deletions">
        <a href="https://github.com/ArduPilot/ardupilot/pull/18137/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+302</span>
          <span class="removed">-117</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">This PR gives a simple way to lower the logging rate for smaller logs. It gives <span class="show-more-container"><a href="https://github.com/ArduPilot/ardupilot/pull/18137" target="_blank" rel="noopener nofollow ugc" class="show-more">…</a></span><span class="excerpt hidden">a single LOG_FILE_RATEMAX parameter for the maximum rate for streaming messages. Later I'd like to add LOG_MAV_RATEMAX and LOG_BLK_RATEMAX. for the other backends.
Special handling is made of multi-instance messages. If the same streaming message ID is sent within the same scheduler tick then it is assumed to be a multi-instance message and the same send/no-send decision is made for all the messages of that type within a tick. This gives good handling of multi-instance messages.

update: now has LOG_MAV_RATEMAX and LOG_BLK_RATEMAX too</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
Cheers, Tridge</p>

**Comment by peterbarker** [comment]
<p>A little more information - day-old pushes to PRs are getting this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/7/d7ac70ca833742a6e7084aeb4e3d969e0cdde8f6.png" data-download-href="/uploads/short-url/uLW6IIpg56AhOk1v70XiqL8b9VY.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7077857/181092462-ce4c1123-b72c-46e5-a3b1-1440a861822a.png" alt="image" data-base62-sha1="uLW6IIpg56AhOk1v70XiqL8b9VY" width="690" height="98" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/7/d7ac70ca833742a6e7084aeb4e3d969e0cdde8f6_2_690x98.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/7/d7ac70ca833742a6e7084aeb4e3d969e0cdde8f6_2_1035x147.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/7/d7ac70ca833742a6e7084aeb4e3d969e0cdde8f6_2_1380x196.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/7/d7ac70ca833742a6e7084aeb4e3d969e0cdde8f6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1450×208 14.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>others are getting this:</p>
<pre><code class="lang-auto">Can't find any online and idle self-hosted or hosted runner in the current repository, account/organization that matches the required labels: 'ubuntu-20.04'
Waiting for a self-hosted or a hosted runner to pickup this job...
</code></pre>
<p>Switching to <code>ubuntu-18.04</code> didn’t help.</p>

**Comment by peterbarker** [comment]
<p>Note this same problem is almost certainly being discussed over here: <a href="https://github.community/t/hosted-runners-not-picking-up-jobs/191663/5" class="inline-onebox">Hosted runners not picking up jobs - #5 by balupton</a></p>

**Comment by tridge** [comment, marked as answer]
<p>the problem has resolved itself</p>
